### PR TITLE
add <comparator for time and datetime

### PR DESCRIPTION
**Pull request by Aiee**
1. added constructor and < comparator for time and datetime
2. updated ValueTest.cpp regarding time and datetime

### CODE REVIEW

File: src/common/datatypes/Date.h [hunk 1]
```diff
@@ -81,6 +81,9 @@ struct Time {
     int8_t sec;
     int32_t microsec;
 
+    Time() : hour{0}, minute{0}, sec{0}, microsec{0} {}
+    Time(int8_t h, int8_t m, int8_t s, int8_t ms) : hour{h}, minute{m}, sec{s}, microsec{ms} {}
```
Comment (jievince) [inline src/common/datatypes/Date.h:85]: microsec is of type int32_t

File: src/common/datatypes/Date.h [hunk 2]
```diff
@@ -112,6 +131,10 @@ struct DateTime {
     int8_t sec;
     int32_t microsec;
 
+    DateTime() : year{0}, month{1}, day{1}, hour{0}, minute{0}, sec{0}, microsec{0} {}
+    DateTime(int16_t y, int8_t mon, int8_t d, int8_t h, int8_t m, int8_t s, int8_t ms) :
```
Comment (jievince) [inline src/common/datatypes/Date.h:135]: ditto

File: src/common/datatypes/Date.h (hunk 1 above)
Comment (Shylock-Hg) [reply src/common/datatypes/Date.h:85]: And it's better to named `us`.

Comment (Aiee) [reply src/common/datatypes/Date.h]: thanks for your feedbacks

**Review by jievince — APPROVED** [review]

File: src/common/datatypes/test/ValueTest.cpp [hunk 3]
```diff
@@ -382,7 +387,32 @@ TEST(Value, Comparison) {
         EXPECT_EQ(Value::Type::BOOL, v.type());
         EXPECT_EQ(true, v.getBool());
     }
+    // time
+    {
+        Value v = vTime1 == vTime2;
```
Comment (Shylock-Hg) [inline src/common/datatypes/test/ValueTest.cpp:392]: Why store the result to Value? I think you can keep it bool.

Comment (Aiee) [reply src/common/datatypes/test/ValueTest.cpp:392]: since all other tests are using the same syntax, I'd like to keep the persistence of code.

Comment (Shylock-Hg) [reply src/common/datatypes/test/ValueTest.cpp:392]: OK

**Review by Shylock-Hg — APPROVED** [review]
LGTM

**Review by dutor — APPROVED** [review]
Well done.

**Pull request merged**